db/multi_cf_iterator.cc

```diff
@@ -9,54 +9,95 @@
 
 namespace ROCKSDB_NAMESPACE {
 
-void MultiCfIterator::SeekToFirst() {
+void MultiCfIterator::SeekCommon(
+    const std::function<void(Iterator*)>& child_seek_func,
```

Comment:
`std::function` does type erasure and has similar overhead to a virtual function call. We could avoid this by turning `SeekCommon` into a member function template (`template<typename ChildSeekFuncType> void MultiCfIterator::SeekCommon(ChildSeekFuncType child_seek_func, ...)`)